### DISCUSSION TITLE
【建议】建议逐步统一、新增、补充完善常用组件各自的生命周期回调函数！

### DISCUSSION THREAD

**Discussion by LSL1618** [layui/layui#1701] [建议]
建议逐步统一、新增、补充完善常用组件各自的生命周期回调函数！组件的生命周期回调函数非常适合处理很多不同的业务场景或需求，从而达到最佳的或满意的使用效果。
比如table组件目前已支持的生命周期回调函数有`before、done、error、complete、parseData`，但是在完整生命周期中理论上不太完善，另外命名不太清楚直观，无法从方法名称上一眼看出来是干什么的，我知道有文档说明，事件或方法命名上建议参考`easyui、topjui、vxe-table`等等。
比如layer组件目前已支持的生命周期回调函数有`success、cancel、end、resizing、full、min、restore`等等，layer组件在生命周期支持完整度上算是最全的。
大部分组件的生命周期一般会有实例渲染前后`onBeforeRender/onAfterRender`、数据加载前后`onBeforeLoad/onAfterLoad`、数据加载成功或失败`onLoadSuccess/onLoadError`、实例销毁前后`onBeforeDestroy/onAfterDestroy`等等通用周期，另外个别组件可能会有它自己独有的特色的生命周期就像layer组件。
这是目前的一个草稿想法，有不对或不足之处大家一起讨论完善！

**Comment by sentsim** [comment]
Layui 组件在规范设计上确实做得不够到位，早期为了实现各组件的功能，而并没有对其进行统筹，导致组件内部各自为战，尽管大体上 API 具有相通点，但个别用法、命名仍然存在一些过于随性的细微差异，代码实现上也存在较多待优化的冗余部分，而这些很难在 2.x 的基础上进行改造，所有一切都必须推到重塑，所以 3.0 要承载的东西比早前预期实在是要多得多。

**Reply by LSL1618** [reply]
2.X为了保证兼容性和稳定性确实不宜大改，所以这些重任就落在3.0上了！

**Reply by lanrenbulan** [reply]
3.0 预计啥时能见面呢
